File: frontend/src/services/user.service.ts
import api from './api';
import { User } from '@/types/auth.types';

export const userService = {
  async getCurrentUser(): Promise<User> {
    const response = await api.get<User>('/users/me');
    return response.data;
  },

  async updateProfile(data: Partial<User>): Promise<User> {
    const response = await api.put<User>('/users/me', data);
    return response.data;
  },

  async changePassword(currentPassword: string, newPassword: string): Promise<void> {
    await api.put('/users/me/password', { currentPassword, newPassword });
  },

  async searchUsers(query?: string): Promise<User[]> {
    const params = query ? { query } : {};
    const response = await api.get<User[]>('/users', { params });
    return response.data;
  },

  async getAllUsers(): Promise<User[]> {
    const response = await api.get<User[]>('/users');
    return response.data;
  },
};

File: backend/src/main/java/kg/taskflow/service/UserService.java
package kg.taskflow.service;

import kg.taskflow.db.entity.User;
import kg.taskflow.dto.user.ChangePasswordRequest;
import kg.taskflow.dto.user.UpdateProfileRequest;
import kg.taskflow.dto.user.UserDto;
import org.springframework.web.multipart.MultipartFile;

import java.util.List;
import java.util.UUID;

public interface UserService {

    User getById(UUID id);

    User getByEmail(String email);

    UserDto getCurrentUser();

    UserDto updateCurrentUserProfile(UpdateProfileRequest request);

    UserDto updateProfile(UUID id, String firstName, String lastName, String phone);

    void changePassword(ChangePasswordRequest request);

    void updateAvatar(UUID id, String avatarUrl);

    String uploadAvatar(MultipartFile file);

    void deleteAvatar();

    List<UserDto> searchUsers(String query);

    List<UserDto> getAllUsers();
}
